hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/metrics/FlinkCompactionMetrics.java

```diff
@@ -103,4 +110,16 @@ public void endCompaction() {
     this.compactionCost = stopTimer(COMPACTION_KEY);
   }
 
+  public void setCompactionFailedState(CompactionState compactionState){
+    this.compactionFailedState = compactionState.state;
+  }
+
+  public enum CompactionState{
+     SUCCESS(0),FAILED(1);
+     int state;
```

Comment:
The metric only has 0 and 1 as values, does it make sense to represent it as a metric?

Reply:
en, this metric works like a state signal used to alert people to know the final compaction state this time. when the mor table happens many rollback, may user can manual operate the hudi mor table or flink job.  the pending count | delay cost may cant't represent the final result.